src/main/java/skinsrestorer/shared/utils/C.java

```diff
@@ -19,6 +19,20 @@ public static boolean validUsername(String username) {
     }
 
     public static boolean validUrl(String url) {
+        return urlPattern.matcher(url).matches()
+                && (url.startsWith("https://i.imgur.com/")
+                || url.startsWith("http://i.imgur.com/")
+                || url.startsWith("i.imgur.com/")
+                || url.startsWith("https://storage.googleapis.com/")
+                || url.startsWith("http://storage.googleapis.com/")
+                || url.startsWith("storage.googleapis.com/")
+                || url.startsWith("https://cdn.discordapp.com/")
+                || url.startsWith("http://cdn.discordapp.com/")
+                || url.startsWith("cdn.discordapp.com/")
```

Comment:
Why not an array with the URL and add URI matches (protocol, "://", "/") using Patterns?